components/overflow-group/overflow-group.js

```diff
@@ -0,0 +1,468 @@
+				type: Number,
+			}
+		};
+	}
+
+	static get styles() {
+		return [offscreenStyles,
+			css`
+			:host {
+				display: block;
+			}
+			:host([hidden]) {
+				display: none;
+			}
+			:host([opener-style="subtle"]) {
+				--d2l-button-icon-fill-color: var(--d2l-color-celestine);
+				--d2l-button-icon-fill-color-hover: var(--d2l-color-celestine-minus-1);
+			}
+			.d2l-overflow-group-container {
+				display: flex;
+				flex-wrap: wrap;
+			}
+			.d2l-overflow-group-container ::slotted(d2l-button),
+			.d2l-overflow-group-container ::slotted(d2l-button-icon),
+			.d2l-overflow-group-container ::slotted(d2l-link),
+			.d2l-overflow-group-container ::slotted(span),
+			.d2l-overflow-group-container ::slotted(d2l-dropdown:not(.d2l-overflow-dropdown)),
+			.d2l-overflow-group-container ::slotted(d2l-dropdown-button),
+			.d2l-overflow-group-container ::slotted(d2l-dropdown-button-subtle),
+			.d2l-overflow-group-container ::slotted(d2l-dropdown-more),
+			.d2l-overflow-group-container ::slotted(d2l-dropdown-context-menu) {
+				margin-right: 0.6rem;
+			}
+			:host([dir="rtl"]) .d2l-overflow-group-container ::slotted(d2l-button),
+			:host([dir="rtl"]) .d2l-overflow-group-container ::slotted(d2l-button-icon),
+			:host([dir="rtl"]) .d2l-overflow-group-container ::slotted(d2l-link),
+			:host([dir="rtl"]) .d2l-overflow-group-container ::slotted(span),
+			:host([dir="rtl"]) .d2l-overflow-group-container ::slotted(d2l-dropdown:not(.d2l-overflow-dropdown)),
+			:host([dir="rtl"]) .d2l-overflow-group-container ::slotted(d2l-dropdown-button),
+			:host([dir="rtl"]) .d2l-overflow-group-container ::slotted(d2l-dropdown-button-subtle),
+			:host([dir="rtl"]) .d2l-overflow-group-container ::slotted(d2l-dropdown-more),
+			:host([dir="rtl"]) .d2l-overflow-group-container ::slotted(d2l-dropdown-context-menu) {
+				margin-left: 0.6rem;
+				margin-right: 0;
+			}
+			:host([opener-style="subtle"]) .d2l-overflow-group-container ::slotted(d2l-button-subtle) {
+				margin-right: 0.2rem;
+			}
+			:host([opener-style="subtle"][dir="rtl"]) .d2l-overflow-group-container ::slotted(d2l-button-subtle) {
+				margin-left: 0.2rem;
+				margin-right: 0;
+			}
+			.d2l-overflow-group-container ::slotted([data-is-chomped]) {
+				display: none !important;
+			}`
+		];
+	}
+	constructor() {
+		super();
+		this._handleResize = this._handleResize.bind(this);
+
+		this._throttledResize = throttle(this._handleResize, 15);
+
+		this._overflowHidden = false;
+		this.autoShow = false;
+		this.maxToShow = -1;
+		this.minToShow = 1;
+		this.openerStyle = OPENER_STYLE.DEFAULT;
+		this.openerType = OPENER_TYPE.DEFAULT;
+		this._mini = this.openerType === OPENER_TYPE.ICON;
+
+		this._slotItems = [];
+	}
+
+	async firstUpdated() {
+		super.firstUpdated();
+
+		// selected elements
+		this._buttonSlot = this.shadowRoot.querySelector('slot');
+
+		this._container = this.shadowRoot.querySelector('.d2l-overflow-group-container');
+
+		this._getItems();
+
+		if (this.autoShow) {
```

Comment:
`_autoDetectBoundaries` gets called in the `updated()` callback, so I'm wondering if this is necessary in `firstUpdated`? Seems like it might get called twice.